crates/router/src/types/api.rs

```diff
@@ -39,10 +39,12 @@ pub trait ConnectorCommon {
     fn id(&self) -> &'static str;
 
     /// HTTP header used for authorization.
+    #[allow(clippy::type_complexity)]
     fn get_auth_header(
         &self,
         _auth_type: &types::ConnectorAuthType,
-    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+    ) -> CustomResult<Vec<(String, masking::Secret<String, masking::ApiKey>)>, errors::ConnectorError>
```

Comment:
I believe using type aliases would eliminate the need for this `allow` attribute.